### PR TITLE
Podcast migration set up om

### LINKED ISSUE #8: Create migration and seeding files for podcast data
using `iTunes` API make a network request to retrieve necessary data and seed the file into the `podcats` table

### PR DESCRIPTION
**Pull request by Atos20**
#### What's this PR do? 

- Creates migration file for `podcasts` table
- Defines `podcasts` schema  initial structure
- Makes a network  request to retrieve information from the iTunes API
- Seeds the information from the iTunes API and inserts it into the `podcasts` table
- it removes unnecessary files not needed for this project
- ☢️.  **locally,  this server runs on localhost/indecision**
#### Where should the reviewer start?  
- `db/seeds/dev/podcasts.js`
- `db/migrations/20201028150700_podcasts.js`

#### How should this be manually tested?  
run `npm run server` to start `nodemon`

#### Any background context you want to provide?  

- This PR is part of the server-side of the `indecision-maker` project

#### What are the relevant tickets?  
closes #8 
#### Screenshots (if appropriate) 
Screen Shot 2020-10-28 at 6 00 34 PM

### CODE REVIEW

File: itunesCall.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const fetch = require("node-fetch");
+
+const getPodcastData = async () => {
+    try {
+		const response = await fetch(`https://itunes.apple.com/search?term=$podcast`, {
+			"method": "GET",
```
Comment (BlakeDonn) [inline itunesCall.js:6]: We all discussed briefly if GET method was inherent on fetch requests, but maybe node-fetch makes that not the case? We can definitely do some research on this to see if this code can be refined or not

File: itunesCall.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	}
+	catch (e) {
+		console.log(e);
+	}
+}
+
+const getAllPodcasts = async () => {
+    const data = await getPodcastData()
+    return data
+  }
```
Comment (BlakeDonn) [inline itunesCall.js:19]: Is there a reason why this is split up like this? Seems to me like we could just invoke getPodcastData() outright in the podcasts.js seed export

**Review by BlakeDonn — APPROVED** [review]
Looks good to me, good job getting the data for our podcast backend seeded from an external source.

File: db/migrations/20201028150700_podcasts.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        table.string('podcast_title');
+        table.string('author');
+        table.string('audience');
+        table.string('genre');
+        table.string('collectionViewUrl');
+        table.integer('episode_count');
+        table.string('release_date');
+        table.string('image_60');
+        table.string('image_100');
+      })
```
Comment (nicolegooden) [inline db/migrations/20201028150700_podcasts.js:17]: Do we need `timestamps` for this table?

File: db/seeds/dev/podcasts.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      image_60: podcast.artworkUrl60,
+      image_100:podcast.artworkUrl100
+    });
+
+}
+
+exports.seed = async (knex) => {
+  try {
+    await knex('podcasts').del()
+    let podcastPromise = await getAllPodcasts()
```
Comment (nicolegooden) [inline db/seeds/dev/podcasts.js:25]: Nice use of the fetch to the API here.

File: itunesCall.js (hunk 1 above)
Comment (nicolegooden) [reply itunesCall.js:6]: yeah, this seems like a new concept.  It's possible that this is directly related to backend...maybe it has always happened behind the scenes from the frontend view.

File: itunesCall.js (hunk 2 above)
Comment (nicolegooden) [reply itunesCall.js:19]: I'm wondering the same thing.  What purpose does `getAllPodcasts()` serve?

**Review by nicolegooden — APPROVED** [review]
Thanks for getting this done, Orlando! Very clean.

File: itunesCall.js (hunk 2 above)
Comment (Atos20) [reply itunesCall.js:19]: I think you are right, this is an unnecessary step, e can definitely refactor this once we put everything together.  As of right now, I created this file to get by but I think I will make more sense to have all API call in one single folder/file

File: db/migrations/20201028150700_podcasts.js (hunk 3 above)
Comment (Atos20) [reply db/migrations/20201028150700_podcasts.js:17]: I thought about that, and we should probably included it.

**Comment by Atos20** [comment]
All thanks to Blake for putting the example together.

File: itunesCall.js (hunk 1 above)
Comment (Atos20) [reply itunesCall.js:6]: When we write the table for Movies we can try without  the `GET` and see what happens.

**Pull request merged**